prime-router/src/main/kotlin/azure/ReportFunction.kt

```diff
@@ -268,7 +290,7 @@ class ReportFunction : Logging {
             .map { ResultDetail.param(ROUTE_TO_PARAMETER, "Invalid receiver name: $it") }
         errors.addAll(receiverNameErrors)
 
-        val clientName = request.headers[CLIENT_PARAMETER] ?: request.queryParameters.getOrDefault(CLIENT_PARAMETER, "")
+        val clientName = extractClientHeader(request)
```

Comment:
No need to have clientName as a variable.  Use inline.

Reply:
I looked at this, and it does get used further down to add additional error messages.